### PR TITLE
Added invisible enemies + Refactor secret passage toggles to reduce code duplication

### PR DESCRIPTION
**Pull request by redknight990**
- Added ITogglelable interface for use with echolocatable objects
- Added an LOS object toggle which uses LOSObjectHider to hide objects
- Added RevealOnEcholocate for ObjectToggle to remove code duplication in PlayerEcholocation
- Added invisible property to enemies 
- Added Enemy Invisible prefab variant with reduced weapon damage

Known bug: 
An enemy dying in a secret passage will cause the secret passage to remain revealed for the rest of the gameplay. This is due to `OnTriggerExit` not being called when a GameObject is destroyed.

### CODE REVIEW

**Review by po-trottier — APPROVED** [review]
image

**Review by emilie-martin — COMMENTED** [review]
ooh this is neat, i like it! just one little thing; if we hit an invisible enemy with a spell, should the enemy appear? i feel like it would be intuitive to do so but i just wanted to know what you think about it! idk if it'd be easy to implement in this PR?

**Review by daniela-venuta — APPROVED** [review]
Nice work on the cleanup! Small comment but approved anyways :)

File: Hush/Assets/Scripts/Enemies/Enemy.cs [hunk 1]
```diff
@@ -32,24 +34,28 @@ public class Enemy : MonoBehaviour, IEnemy
...
         [SerializeField] private Animator animator;
         [SerializeField] private SphereCollider visionCollider;
-
+        
         #endregion
     
         #region Private Variables
-
+        
+        private const float BackstabDamageModifier = 2.0f;
```
[the remark this answers is not in the document]
Comment (redknight990) [reply Hush/Assets/Scripts/Enemies/Enemy.cs]: Changed in 77ce0ff

**Comment by redknight990** [comment]
> ooh this is neat, i like it! just one little thing; if we hit an invisible enemy with a spell, should the enemy appear? i feel like it would be intuitive to do so but i just wanted to know what you think about it! idk if it'd be easy to implement in this PR?

Added in 77ce0ff

File: Hush/Assets/Scripts/Enemies/Enemy.cs [hunk 2]
```diff
@@ -354,6 +378,13 @@ private void SetState(EnemyState state)
                 GameManager.Instance.EnemyStoppedAttacking(ID);
             }
         }
+        
+        private IEnumerator Reveal()
+        {
+            Show();
+            yield return new WaitForSeconds(1.0f);
```
Comment (po-trottier) [inline Hush/Assets/Scripts/Enemies/Enemy.cs:385]: Would be cool if this was a serializeField

**Review by po-trottier — APPROVED** [review]
noice!

**Review by daniela-venuta — APPROVED** [review]

File: Hush/Assets/Scripts/Enemies/Enemy.cs (hunk 2 above)
Comment (redknight990) [reply Hush/Assets/Scripts/Enemies/Enemy.cs]: Changed in acb305f

**Review by daniela-venuta — APPROVED** [review]

**Pull request merged**